lib/puppet/type/cert_key_bundle.rb

```diff
@@ -0,0 +1,96 @@
+Puppet::Type.newtype(:cert_key_bundle) do
+  desc 'combines a certificate and private key into a single file'
+
+  ensurable
+
+  newparam(:path, :namevar => true) do
+    desc "Path to the file that will contain the certificate and private key bundle"
+    isrequired
+  end
+
+  newparam(:certificate) do
+    desc "Path to certificate file to include in bundle"
+    isrequired
+
+    def insync?(is)
+      File.read(self[:path]).include?(File.read(should))
+    end
+  end
+
+  newparam(:private_key) do
+    desc "Path to private key file to include in bundle"
+    isrequired
+
+    def insync?(is)
+      should = format_as_pkcs_1(should) if self[:force_pkcs_1]
+      File.read(self[:path]).include?(File.read(should))
+    end
+  end
+
+  newparam(:owner, parent: Puppet::Type::File::Owner) do
+    desc "Specifies the owner of the file. Valid options: a string containing a username or integer containing a uid."
+  end
+
+  newparam(:group, parent: Puppet::Type::File::Group) do
+    desc "Specifies a permissions group for the file. Valid options: a string containing a group name or integer containing a gid."
+  end
+
+  newparam(:mode, parent: Puppet::Type::File::Mode) do
+    desc "Specifies the permissions mode of the file. Valid options: a string containing a permission mode value in octal notation."
+  end
+
+  newparam(:force_pkcs_1, :boolean => true, :parent => Puppet::Parameter::Boolean) do
+    desc "Ensures the private key is in PKCS#1 format. This is required for services like Apache reverse proxy"
+  end
```

Comment:
```suggestion
  newparam(:force_pkcs_1, :boolean => true, :parent => Puppet::Parameter::Boolean) do
    desc "Ensures the private key is in PKCS#1 format. This is required for services like Apache reverse proxy"
    defaultto true
  end
```

We hardcode `force_pkcs_ => true` in the manifest, so it could be dropped there if we set the default here.

I'm not sure what is the circumstance where we don't want this set to true.

Reply:
I was going to go see if this was still a bug on EL8 with FIPS enabled, which led to find other unrelated problems with FIPS and crypto. Hooray!

My intent was to say we should only set this to true in certain circumstances. And I think generally that is my feeling on this, I would rather make this obvious that we are setting it to true, rather than hide it as the default since this is a workaround and should not be the default behavior.

Reply:
Yea, so after some FIPS investigation I still lean towards not defaulting to this behavior for this provider and having to specify it given its only required for the specific bundled use of Apache. There could be future use cases where we want to bundle cert/key and I'd prefer to default to not forcing this.

Reply:
thanks @ehelms , leaving it as-is works for me then. BTW, if there are any PRs or anything in writing related to the further issues you discovered, please link and I'd love to read up on it later

Reply:
Sure, here are some of the FIPS fixes I found needed https://github.com/Katello/katello-certs-tools/pull/36